### PR TITLE
iOS: Fix statusbar style handling (#56 / #13)

### PR DESCRIPTION
**Pull request by kbshl**
Hey @viezel,

seems like I finally found the error that is causing to ignore statusbar style changes for the center window.

TIL
Cause the center window never gets added to the `containedWindows`
property on TiRootViewController, statusbar style changes on center
window are ignored. More specifically the left/rights drawers window is
asked for the statusbar style by TiRootViewController.
This is fixed by calling `windowWillOpen` and `windowDidOpen` on the
center window proxy AFTER fetching the view controllers for the left and
right drawer. Cause fetching the left and right drawer view controllers
is implicitly calling `windowWillOpen` and `windowDidOpen` on their
proxies, adding them to the TiRootViewControllers `containedWindows`
array.

Done some more stuff:
- [x] Add default LICENSE file
- [x] Add pragma marks to drawer view class
- [x] Add preferredStatusBarStyle and childViewControllerForStatusBarStyle handling to drawer proxy
- [x] Remove unused code
- [x] Remove log call on module load
- [x] Add class name to log calls
- [x] Add windowWillClose cleanup call to previous navigation controller on center window replacement
- [x] Fix preferredStatusBarStyle handling for NappDrawer view. Define NappDrawer view preferred statusbar style through CustomMMDrawerController 
- [x] Disable child view controller consulting for statusbar style on NappDrawer view preferredStatusBarStyle call
- [x] Fix statusBarStyle handling for center window
- [x] Lint source code through clang-format
- [x] Bump iOS module version to 2.1.1
- [x] Add dist zip file for version 2.1.1

Likely to fix:
- #56 
- #13

Binary is added as an commit.

### CODE REVIEW

**Comment by viezel** [comment]
@hansemannn can you review this?

File: ios/Classes/DkNappDrawerDrawer.m [hunk 1]
```diff
@@ -100,23 +104,59 @@ - (MMDrawerController *)controller
...
+
       controller = [[CustomMMDrawerController alloc] initWithCenterViewController:centerWindow
-                                                        rightDrawerViewController:ControllerForViewProxy(rightWindow)];
-    } else {
+                                                        rightDrawerViewController:rightController];
+
       //error
-      NSLog(@"[ERROR] NappDrawer: No windows assigned");
+    } else {
+      NSLog(@"[DkNappDrawerDrawer][ERROR] No windows assigned");
```
Comment (hansemannn) [inline ios/Classes/DkNappDrawerDrawer.m]: For proper error reporting on the console, the `[ERROR]` prefix should be first.

File: ios/Classes/DkNappDrawerDrawer.m [hunk 2]
```diff
@@ -223,12 +263,24 @@ - (void)setCenterWindow_:(id)args
     useNavController = YES;
   }
   UIViewController *centerWindow = useNavController ? NavigationControllerForViewProxy([self.proxy valueForUndefinedKey:@"centerWindow"]) : ControllerForViewProxy([self.proxy valueForUndefinedKey:@"centerWindow"]);
+
+  if (useNavController) {
+
+    TiUIiOSNavWindowProxy *centerProxy = [self.proxy valueForUndefinedKey:@"centerWindow"];
```
Comment (hansemannn) [inline ios/Classes/DkNappDrawerDrawer.m]: Remove empty line L266

File: ios/Classes/DkNappDrawerDrawer.m [hunk 3]
```diff
@@ -223,12 +263,24 @@ - (void)setCenterWindow_:(id)args
...
+
+    if (controller) {
```
Comment (hansemannn) [inline ios/Classes/DkNappDrawerDrawer.m]: `controller != nil` for style

File: ios/Classes/DkNappDrawerDrawer.m [hunk 4]
```diff
@@ -223,12 +263,24 @@ - (void)setCenterWindow_:(id)args
...
+    }
+  }
+
   [controller setCenterViewController:centerWindow];
 
   // Cleanup
   if (useNavController) {
     if (navProxy != nil) {
-      NSLog(@"[DEBUG| Destroying old nav proxy");
+      NSLog(@"[DkNappDrawerDrawer][INFO] Destroying old nav proxy");
```
Comment (hansemannn) [inline ios/Classes/DkNappDrawerDrawer.m]: This should not be logged at all.

File: ios/Classes/DkNappDrawerDrawerProxy.m [hunk 5]
```diff
@@ -53,6 +53,16 @@ - (TiUIView *)newView
   return [[DkNappDrawerDrawer alloc] init];
 }
 
+- (UIStatusBarStyle)preferredStatusBarStyle
+{
+  [[self _controller] preferredStatusBarStyle];
```
Comment (hansemannn) [inline ios/Classes/DkNappDrawerDrawerProxy.m]: I am wondering how this can work? The `return` is missing.

File: ios/LICENSE [hunk 6]
```diff
@@ -0,0 +1,219 @@
+Copyright 2013 Appcelerator, Inc.
```
Comment (hansemannn) [inline ios/LICENSE]: Why not use the [existing license](https://github.com/viezel/NappDrawer/blob/master/LICENSE) in the top-level directory? The copyright is also not at appc but Mads.

Comment (kbshl) [reply ios/LICENSE]: You absolutely right, I missed the top level license 🤦🏻‍♂️

**Comment by kbshl** [comment]
@hansemannn, thank you for your time and the review 🙇🏻‍♂️

Made the changes you are mentioned and tested it. Hope it fits now.

**Comment by fahad86** [comment]
How do I configure the preferred Status bar color? 

I tried adding this param:

    preferredStatusBarStyle = Ti.UI.iOS.StatusBar.LIGHT_CONTENT;

**Comment by fahad86** [comment]
How can I set the statusBarStyle?

I tried:

`NappDrawerModule.setStatusBarStyle(NappDrawerModule.STATUSBAR_WHITE);`

but no use

**Pull request merged**